### PR TITLE
chore: add benchmarks

### PR DESCRIPTION
**Pull request by H4ad**
I had these benchmarks from the last time I did some perf research in this repo, so I thought maybe it was a good idea to have them in the repo itself, so other people can use them and you guys can verify if the perf is still good or you had any regression.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -37,3 +37,4 @@ tap-testdir*/
 !/tap-snapshots/
 !/test/
 !/tsconfig.json
+!/benchmarks
```
[the remark this answers is not in the document]
Comment (wraithgar) [reply .gitignore:40]: I believe `templateOSS.allowPaths` is the package.json entry that affects this.

File: benchmarks/package.json [hunk 2]
```diff
@@ -0,0 +1,13 @@
+{
```
Comment (wraithgar) [inline benchmarks/package.json:1]: Adding this feels way more cumbersome than just adding `benchmark` as a devDependency.

File: package.json [hunk 3]
```diff
@@ -71,7 +72,8 @@
       "/ranges/",
       "/index.js",
       "/preload.js",
-      "/range.bnf"
+      "/range.bnf",
+      "/benchmarks"
```
Comment (wraithgar) [inline package.json:76]: `npm run template-oss-apply` will apply this and rebuild the .gitignore.

File: benchmarks/.gitignore [hunk 4]
```diff
@@ -0,0 +1 @@
+node_modules
```
Comment (wraithgar) [inline benchmarks/.gitignore:1]: Don't need this file or the empty README.md

File: package.json (hunk 3 above)
Comment (H4ad) [reply package.json:76]: I ran but I removed the changes on `.github/workflows` about erasing a folder called `release`.

Comment (wraithgar) [reply package.json:76]: your local version may have been out of date. looks like the linting is green now which means we should be good now.

**Review by wraithgar — APPROVED** [review]

**Pull request merged**